module/VuFind/src/VuFind/RecordDriver/DefaultRecord.php

```diff
@@ -281,6 +281,54 @@ public function getCleanISBN()
         return $isbn13;
     }
 
+    /**
+     * Return all ISBNs found in the record.
+     *
+     * @param array $userFlags Flags to control behavior:
+     *  - 'only10' returns only ISBN-10s
+     *  - 'prefer10' returns ISBN-10s if available, otherwise ISBN-13s (default)
+     *  - 'normalize13' returns ISBN-13s, normalizing ISBN-10s to ISBN-13s
+     *  - 'filterInvalid' filters out invalid ISBNs
+     * Only one of 'only10', 'prefer10', or 'normalize13' should be set to true.
```

Comment:
Should we add some logic to enforce this, like:

```
$exclusiveFlags = ($flags['only10'] ? 1 : 0) + ($flags['prefer10'] ? 1 : 0) + ($flags['normalize13'] ? 1 : 0);
if ($exclusiveFlags > 1) {
    throw new \Exception('Invalid combination of flags provided.');
}
````

Reply:
I refactored the parameters a bit - now there is no flags array but two params - mode (only10, prefer10, normalize13) and one boolean $filterInvalid